forte/helpers/helpers.cc

```diff
@@ -138,6 +139,84 @@ std::pair<double, std::string> to_xb(size_t nele, size_t type_size) {
     return out;
 }
 
+void matrix_transpose_in_place(double* data, const size_t m, const size_t n) {
+    int nthreads = std::min(omp_get_max_threads(), int(m > n ? n : m));
+    std::vector<double> tmp(nthreads * (m > n ? m : n));
+    double* tmp_ptr = tmp.data();
```

Comment:
I am not a big fan of the raw pointer here. Can this be avoided?

Reply:
This function rearranges the data in memory, which is somewhat dangerous. I use it to trick ambit to do in-place transpose of 3-index integrals. Forte uses Qpq order but a more convenient order would be pqQ for DF-PT2. Anyways, I think the raw pointer is more general, but I guess vector<double> works fine as well.

Reply:
If you are referring line 145, I think it is the most convenient way for lines like 155 and 160.

Reply:
Changed to iterator.

Reply:
I thought to put it in Psi4 because in-place matrix transposition is not available. I think there are better algorithm out there (e.g., https://onlinelibrary.wiley.com/doi/10.1002/cpe.5071) but the current one does the job.